ArduPlane/RC_Channel.cpp

```diff
@@ -164,6 +168,26 @@ void RC_Channel_Plane::do_aux_function(const aux_func_t ch_option, const AuxSwit
         do_aux_function_q_assist_state(ch_flag);
         break;
 
+    case AUX_FUNC::TER_DISABLE:
+            switch (ch_flag) {
+            case AuxSwitchPos::HIGH:
+                plane.non_auto_terrain_disable = true;
+                if (plane.control_mode == &plane.mode_fbwb || plane.control_mode == &plane.mode_cruise) {
```

Comment:
using the new accessor this can be:
if (plane.control_mode->allows_terrain_disable()) {

Reply:
thanks...done